src/wfs/src/main/java/org/geoserver/wfs/json/GeoJSONBuilder.java

```diff
@@ -297,11 +301,17 @@ public static int getGeometryType(Geometry geometry) {
     public GeoJSONBuilder value(Object value) {
         if (value instanceof java.util.Date || value instanceof Calendar) {
             value = Converters.convert(value, String.class);
+        } else if (value instanceof Double) {
+            value = roundToNumberOfDecimalPlaces((Double)value);
         }
         super.value(value);
         return this;
     }
-    
+
+    private double roundToNumberOfDecimalPlaces(Double value) {
+        return new BigDecimal((Double)value).setScale(numDecimals, BigDecimal.ROUND_HALF_UP).doubleValue();		
```

Comment:
append(formattedStringValue) is private in json-lib but the bare writer is
protected, so I would have to call that directly.  That strikes me as a
little fragile, but I can do it if you insist.

Also json-lib handles removing trailing zeros so we would have to duplicate
that code to be consistent.

My suggestion would be to just change the writeCoordinate to do the
rounding but still call super.value() on the resulting double.  The Double
wrapper is going to be much less significant than the StringBuffer
allocation and the other work that is going on there.

On Sat, Jan 9, 2016 at 8:18 PM, Andrea Aime <notifications@github.com>
wrote:

> In src/wfs/src/main/java/org/geoserver/wfs/json/GeoJSONBuilder.java
> <https://github.com/geoserver/geoserver/pull/1392#discussion_r49262804>:
>
> >          }
> >          super.value(value);
> >          return this;
> >      }
> > -
> > +
> > +    private double roundToNumberOfDecimalPlaces(Double value) {
> > +        return new BigDecimal((Double)value).setScale(numDecimals, BigDecimal.ROUND_HALF_UP).doubleValue();		
>
> For best performance, we could also dodge the jsonlib creation of a Double
> wrapper, override value(double d) and make it call directly
> append(formattedStringValue) instead.
>
> As a separate observation, your change has the side effect of also
> formatting attributes that happen to be doubles, which we might need to
> preserve with full accuracy. Probably best to change writeCoordinate to
> call onto an optimized writeOrdinate method that formats the double the
> efficient way, and then appends a string
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/geoserver/geoserver/pull/1392/files#r49262804>.
>